docs/examples/yang19.ipynb

```diff
@@ -0,0 +1,573 @@
+    "    plot_performance=True,\n",
+    "    fig_kwargs={'figsize': (9, 5)},\n",
+    ")\n",
+    "\n",
+    "for text_obj in fig.findobj(match=plt.Text):\n",
+    "    text_obj.set_fontsize(11)\n",
+    "fig.savefig(f\"untrained_{task}.pdf\", dpi=300, bbox_inches='tight')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We wrap the `yang19` environments using NeuroGym’s `ScheduleEnvs` wrapper, which allows us to interleave multiple tasks into a single training loop. Each cognitive task is instantiated as a separate environment, and a scheduling policy (`RandomSchedule`) determines which task is sampled on each trial. When `env_input=True`, a one-hot vector is appended to the observation to indicate the currently active task, allowing the agent to learn task-specific behavior within a unified architecture.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Environment parameters\n",
+    "# This setting is low to speed up testing, we recommend setting it to at least 1000\n",
+    "EVAL_TRIALS = 100\n",
+    "kwargs = {'dt': 100, 'dim_ring': 2}\n",
+    "seq_len = 100\n",
+    "\n",
+    "# Make supervised dataset\n",
+    "tasks = ngym.get_collection('yang19')\n",
+    "envs = [gym.make(task, **kwargs) for task in tasks]\n",
+    "schedule = RandomSchedule(len(envs))\n",
+    "env = ScheduleEnvs(envs, schedule=schedule, env_input=True)\n",
+    "\n",
+    "# Print environment specifications\n",
+    "print(\"Trial timing (in milliseconds):\")\n",
+    "print(env.timing)\n",
+    "\n",
+    "print(\"\\nObservation space structure:\")\n",
+    "print(env.observation_space)\n",
+    "\n",
+    "print(\"\\nAction space structure:\")\n",
+    "print(env.action_space)\n",
+    "print(\"Action mapping:\")\n",
+    "print(env.action_space.name)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 1.1.3 Random Agent Behavior\n",
+    "\n",
+    "Let's now plot the behavior of a random agent on the task. The agent will randomly choose between the two options (left/right, blue line), and we will visualize its behavior over 5 trials. We will also plot the reward received by the agent at each time step, as well as the performance on each trial. Note that performance is only defined at the end of a trial: it is 1 if the agent made the correct choice, and 0 otherwise.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "obs, info = env.reset()\n",
+    "\n",
+    "# Visualize example trials\n",
+    "from neurogym.utils.plotting import plot_env as plot_env_orig\n",
+    "\n",
+    "fig = plot_env_orig(\n",
+    "    env,\n",
+    "    name='Yang et al.',\n",
+    "    ob_traces=None,\n",
+    "    num_trials=5,\n",
+    "    plot_performance=True,\n",
+    "    fig_kwargs={'figsize': (9, 5)},\n",
+    ")\n",
```

Comment:
This command fails sometimes. Running it locally, it works in some sometimes but not always. In the notebook workflow it failed as well. 

Could this be related to the problem you mentioned in a different PR of having only catch trials on low trial counts? If so, maybe it would be better to catch errors for these instances and return a better error message than the one now seen.

Reply:
I tested it and it is related to low number of trials. I've created #245 to deal with these kinds of issues. Rebasing (or merging) off that branch should solve this issue.